hal/sparcv8leon3/gr716/timer.c

```diff
@@ -0,0 +1,157 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * Timer controller
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+
+/* Timer control bitfields */
+#define TIMER_ENABLE      (1)
+#define TIMER_ONESHOT     (0 << 1)
```

Comment:
Such defines are somewhat dangerous and confusing, as we most commonly use them as:
`*reg |= DEFINE;`

This would result in case of `(0 << 1)` in
`*reg |= 0;`
no operation